wandb/sdk/internal/sender.py

```diff
@@ -603,6 +608,9 @@ def _save_history(self, history_dict):
     def send_history(self, data):
         history = data.history
         history_dict = proto_util.dict_from_proto_list(history.item)
+        if self._settings._sync_tensorboard:
+            history_dict["_step"] = self._step
+            self._step += 1
```

Comment:
There was a rough design mentioned at:
https://weightsandbiases.slack.com/archives/C0108SLEP6K/p1611274855028500?thread_ts=1611272615.015700&cid=C0108SLEP6K

My reading of the proposal was that all auto incremented step calculations would move to the internal process.

Also, as i commented in the above thread I think it needs to happen in the handler thread, not the sender.  handler thread is run all the time, even in offline mode.

Note that we do similar type logic for config and summary in the handler thread.   history was the next one to get this functionality.

Reply:
I can move it over to handler